File: cpp/hot100/10.和为 K 的子数组.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> map;
        map[0]=1;
        int ans=0,sum=0;
        for(auto num:nums){
            sum+=num;
            if(map.find(sum-k)!=map.end()){
                ans+=map[sum-k];
            }
            map[sum]++;
        }
        return ans;
    }
};

File: python/hot100/10.和为 K 的子数组.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        counts=defaultdict(int)
        counts[0]=1
        sum,ans=0,0
        for num in nums:
            sum+=num
            ans+=counts[sum-k]
            counts[sum]+=1
        return ans